### ISSUE TITLE
Feat: Add support for camelCase types

### ISSUE THREAD

**Issue by MariuzM** [drizzle-team/drizzle-kysely#2]
When i use Kysely + `plugins: [new CamelCasePlugin()]` i get types that are `camelCase`, but when i also want to use `Kyselify` types are only `snake_case`. 

Any chance to have support for camelCase when using with `plugins: [new CamelCasePlugin()]`?

```tsx
import type { Kyselify } from 'drizzle-orm/kysely'
import { CamelCasePlugin, Kysely, PostgresDialect } from 'kysely'
import { Pool } from 'pg'

import type { UsersTable } from '-db/index.tables'

export interface DB {
	users: Kyselify<typeof UsersTable>
}

export const db = new Kysely<DB>({
	dialect: new PostgresDialect({ pool: new Pool({ connectionString: process.env.DATABASE_URL }) }),
	plugins: [new CamelCasePlugin()],
})

export const main = async () => {
	const t1 = await db.selectFrom('users').select(['']).execute()
	t1[0]?.accountType === 'host'
	t1[0]?.console.log('🚀 ~ t1:', t1)

	db.insertInto('users')
		.values({
			accountType: 'host',
		})
		.execute()
}
```

<img width="577" alt="image" src="https://github.com/drizzle-team/drizzle-kysely/assets/36937039/e7afd1fd-bb4c-4e31-9925-fb30fdeb5d70">

**Comment by mjbergman92** [comment]
Out of pure curiosity, I have been attempting to define a way to provide Kysely with aliases for column names, and also possibly for table names. 

I have drizzle setup for a postgres database and I am using kysely for a mssql database, and I'm now using dremio to be able to do joins between the two databases and using Kysely to build the queries for dremio.

I'm able to construct the postgres tables using Kyselify, but not with the drizzle aliases. For clarity, I want the names in code to remain the same as when I'm using drizzle.

TLDR:
A custom plugin for Kysely to provide aliases is in store. If I make headway on one, I will share it with the community.

EDIT: 
I took the time to look into making a plugin. The benefits of Kysely seem to also be its weakness in this case. The closest you can get is creating a transformer for ReferenceNode; however, the original table name of aliased tables in not included in the ReferenceNode. That's the missing link here is somehow gaining the context for aliased tables which would then allow you to search a provided alias map to the plugin for the table and column name.

**Comment by aashishpanchal** [comment]
I solve this problem to using own type utility types:

```ts
// types/utility.ts

import {ColumnType} from 'kysely';

// Convert snake_case to camelCase recursively
export type SnakeToCamelCase<S extends string> =
  S extends `${infer Head}_${infer Tail}`
    ? `${Head}${Capitalize<SnakeToCamelCase<Tail>>}`
    : S;

// Column type
type IsColumnType<T> = T extends ColumnType<any, any, any> ? true : false;

// Deeply converts object keys to camelCase (optional)
export type DeepCamelCaseKeys<T> = {
  [K in keyof T as K extends string
    ? SnakeToCamelCase<K>
    : K]: T[K] extends Date
    ? T[K]
    : IsColumnType<T[K]> extends true
      ? T[K] // don't recurse into ColumnType
      : T[K] extends object
        ? DeepCamelCaseKeys<T[K]>
        : T[K];
};
```

```ts
// types/kysely.ts

import type * as tables from '@/db/tables';
import type {Kyselify} from 'drizzle-orm/kysely';
import type {DeepCamelCaseKeys, SnakeToCamelCase} from './utility';

// Map Drizzle table names (snake_case) to Kysely-compatible types
type RawTableMap = {
  [K in keyof typeof tables as (typeof tables)[K]['_']['name']]: Kyselify<
    (typeof tables)[K]
  >;
};

// Final exported type:
// - Table names converted to camelCase
// - Column names converted to camelCase inside each table
export type Database = {
  [K in keyof RawTableMap as SnakeToCamelCase<K & string>]: DeepCamelCaseKeys<
    RawTableMap[K]
  >;
};
```

```ts
// db/client.ts

import pg from 'pg';
import conf from '@/conf';
import type {Database} from '@/types/kysely';
import {CamelCasePlugin, Kysely, PostgresDialect} from 'kysely';

// Create and configure PG pool
export const pool = new pg.Pool({connectionString: process.env.DATABASE_URL!});
// Connect to the database
pool.on('connect', () => console.log('Connected to the database'));
// Graceful shutdown
const shutdown = async () => {
  console.log('Closing database connection...');
  await pool.end();
  process.exit(0);
};
// Handle Ctrl+C
process.on('SIGINT', shutdown);
process.on('SIGTERM', shutdown);

// Initialize Kysely and check connection
const dialect = new PostgresDialect({pool});
export const db = new Kysely<Database>({
  dialect,
  log: conf.IS_DEV ? ['query'] : undefined,
  plugins: [new CamelCasePlugin()],
});
export default db;
// Export types
export type DB = Kysely<Database>;
```

<img width="621" alt="Image" src="https://github.com/user-attachments/assets/6d771969-ab53-4b6d-8fa0-7d5efe0ad3e3" />
